### PR TITLE
xplat: fix lto.gold segfault debian

### PR DESCRIPTION
**Pull request by obastemur**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -93,6 +93,8 @@ WB_CHECK=
 WB_ANALYZE=
 WB_ARGS=
 TARGET_PATH=0
+# -DCMAKE_EXPORT_COMPILE_COMMANDS=ON useful for clang-query tool
+CMAKE_EXPORT_COMPILE_COMMANDS="-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
```
Comment (jianchun) [inline build.sh]: >CMAKE_EXPORT_COMPILE_COMMANDS [](start = 0, length = 29)

You can delete this "CMAKE_EXPORT_COMPILE_COMMANDS"... Too little chance anybody will ever use it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -305,15 +305,14 @@ endif(CLR_CMAKE_PLATFORM_XPLAT)
 
 if (ENABLE_FULL_LTO_SH OR ENABLE_THIN_LTO_SH)
     if (CC_TARGET_OS_LINUX)
-        set(CC_LTO_ENABLED
-            -fuse-ld=gold
-            -Xlinker -plugin=${CHAKRACORE_BINARY_DIR}/../../cc-toolchain/build/lib/LLVMgold.so
-            )
+        set(CC_LTO_ENABLED -use-gold-plugin)
+        set(CC_LTO_ENABLED_C -c)
```
Comment (jianchun) [inline CMakeLists.txt]: >-c [](start = 29, length = 2)

Curious, why "-c" (only run preprocess, compile and assemble steps)?

**Comment by jianchun** [comment]
:shipit:

File: CMakeLists.txt (hunk 2 above)
Comment (obastemur) [inline CMakeLists.txt]: to create LLVM bitcode files

File: build.sh (hunk 1 above)
Comment (obastemur) [inline build.sh]: Updated it anyway

**Comment by obastemur** [comment]
@jianchun Thanks for the review!

**Pull request merged**